src/main/java/eu/darkbot/api/entities/Entity.java

```diff
@@ -0,0 +1,88 @@
+package eu.darkbot.api.entities;
+
+import eu.darkbot.api.entities.other.Effect;
+import eu.darkbot.api.objects.Locatable;
+import eu.darkbot.api.objects.LocationInfo;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collection;
+
+/**
+ * Generic entity object in-game.
+ */
+public interface Entity extends Locatable {
+
+    /**
+     * @return id of the {@link Entity}
+     */
+    int getId();
+
+    /**
+     * @return true if and only if {@link Entity} is loaded in-game and isn't removed
+     * @see #isRemoved()
+     */
+    boolean isValid();
+
+    /**
+     * Checks that entity is removed.
+     * If (isRemoved() == true) then this entity wont be updated anymore.
+     */
+    boolean isRemoved();
```

Comment:
Are these 2 methods complementary? is one the other negated?

If they are the same negated, remove one of the 2 or make it very clear that they are the same
If they are slightly different that one negated of the other, explain the difference in the doc